src/OpenTelemetry.Exporter.Prometheus/Implementation/PrometheusExporterMiddleware.cs

```diff
@@ -79,7 +79,9 @@ public async Task InvokeAsync(HttpContext httpContext)
                     }
                     else
                     {
-                        throw new InvalidOperationException("Collection failure.");
+                        // It's not expected to have no metrics to collect, but it's not necessarily a failure, either.
+                        response.StatusCode = 204;
+                        PrometheusExporterEventSource.Log.NoMetrics();
```

Comment:
an alternative to this would be to just Log "PrometheusExporter return with statuscode {code}", actualCode - and do it for other case as well..

(not a blocker for this PR.)

Reply:
Depends on whether you want "no metrics" to be a warning. Right now it is, which likely isn't the level you want a general response code to be logged at.